File: src/Test/java/com/nishant/algorithms/Sorting/SelectionSortTest.java
package com.nishant.algorithms.sorting;

import com.nishant.algorithms.sorting.selectionsort.SelectionSort;
import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertArrayEquals;

public class SelectionSortTest {

  private int[] arr;

  @Before
  public void setup() {
    arr = new int[]{5, 3, 2, 1, 5};
  }

  @Test
  public void test() {
    SelectionSort.sort(arr);
    assertArrayEquals(new int[]{1, 2, 3, 5, 5}, arr);
  }
}

File: src/test/cpp/sorting/selectionsort/selection_sort_test.cpp
//
// Created by nisha on 6/2/2020.
//

#include "gtest/gtest.h"
#include "../../../../main/cpp/sorting/selectionsort/selection_sort.h"

TEST(Sorting, SelectionSortTest) {
    for(int i = 0; i < 1000; i ++) {
        std::vector<int> arr;

        for(int j = 0; j < 500; j ++) {
            arr.push_back(rand() % 1000);
        }

        std::vector<int> temp = arr;
        std::sort(temp.begin(), temp.end());
        selection_sort(arr);
        ASSERT_EQ(temp, arr);
    }
}